src/main/java/duke/command/recipecommands/AddRecipeIngredientCommand.java

```diff
@@ -0,0 +1,89 @@
+package duke.command.recipecommands;
+
+import duke.command.CommandRecipeIngredient;
+import duke.list.recipelist.RecipeIngredientList;
+
+import duke.storage.RecipeIngredientStorage;
+import duke.ui.Ui;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+
+import static duke.common.Messages.*;
+import static duke.common.RecipeMessages.*;
+
+public class AddRecipeIngredientCommand extends CommandRecipeIngredient {
+
+    public AddRecipeIngredientCommand(String userInput) {
+        this.userInput = userInput;
+    }
+
+    @Override
+    public ArrayList<String> execute(RecipeIngredientList recipeIngredientList, Ui ui, RecipeIngredientStorage recipeIngredientStorage) throws ParseException {
+        ArrayList<String> arrayList = new ArrayList<>();
+        String recipeIngredientWeight = "";
+        String recipeIngredientQuantity = "";
+        String recipeIngredientName = "";
+        String recipeIndex = "";
+        if (userInput.trim().equals(COMMAND_ADD_RECIPE_INGREDIENT)) {
+            arrayList.add(ERROR_MESSAGE_GENERAL + MESSAGE_FOLLOWUP_NUll);
+            System.out.println("stuck here1");
+        } else if (userInput.trim().charAt(19) == ' ') {
+            String description = userInput.split("\\s", 2)[1].trim();
+            if (description.contains("q/")) {
+                String temp = description.split("q/", 2)[0].trim();
+                recipeIndex = temp.split("\\s", 2)[0].trim();
+                if (isParsableInt(recipeIndex)) {
+                    recipeIngredientName = temp.split("\\s", 2)[1].trim();
+                    String remaining = description.split("q/", 2)[1].trim();
+                    if (remaining.contains("w/")) {
+                        recipeIngredientQuantity = remaining.split("w/")[0].trim();
+                        recipeIngredientWeight = remaining.split("w/")[1].trim();
+                        if (isParsableDouble(recipeIngredientQuantity)) {
```

Comment:
Five levels of nested if statements are here. Please think of an effective way of refactoring it to make it more readable